File: static/CWE-798/java/vulnerable.java
// CWE-798: Use of Hard-coded Credentials
// A hard-coded database password is embedded in source and used to authenticate.
// Vulnerable sink: DriverManager.getConnection(url, user, "hardcodedPassword")

import java.sql.Connection;
import java.sql.DriverManager;

class CWE_798 {
    private static final String DB_PASSWORD = "S3cr3tP@ssw0rd!";

    public Connection connect(String userInput) throws Exception {
        String url = "jdbc:mysql://localhost/" + userInput;
        return DriverManager.getConnection(url, "admin", DB_PASSWORD);
    }
}

File: static/CWE-798/csharp/vulnerable.cs
// CWE-798: Use of Hard-coded Credentials
// A hard-coded database password is embedded in the connection string.
// Vulnerable sink: new SqlConnection("...;Password=<hardcoded>;")

using System.Data.SqlClient;

namespace GuineaPig.Sast
{
    public class CWE_798
    {
        public void Connect(string userInput)
        {
            string connectionString =
                "Server=db.internal;Database=app;User Id=admin;Password=P@ssw0rd123!;";
            using (var conn = new SqlConnection(connectionString))
            {
                conn.Open();
                using (var cmd = new SqlCommand("SELECT 1 WHERE Tag = '" + userInput + "'", conn))
                {
                    cmd.ExecuteNonQuery();
                }
            }
        }
    }
}
